ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -150,373 +146,63 @@ newtype ByronHash = ByronHash { unByronHash :: CC.Block.HeaderHash }
 convertSlot = coerce
 
-{-------------------------------------------------------------------------------
-  Condense instances
--------------------------------------------------------------------------------}
-
-instance Condense ByronBlock where
-  condense blk =
-      "(header: " <> condensedHeader <>
-      ", body: "  <> condensedBody   <>
-      ")"
-    where
-      condensedHeader = condense
-                      . getHeader
-                      $ blk
-      condensedBody = T.unpack
-                    . sformat build
-                    . CC.UTxO.txpTxs
-                    . CC.Block.bodyTxPayload
-                    . CC.Block.blockBody
-                    . unByronBlock
-                    $ blk
-
-instance Condense (Header ByronBlock) where
-  condense hdr =
-      "(hash: "          <> condensedHash        <>
-      ", previousHash: " <> condensedPrevHash    <>
-      ", slot: "         <> condensedSlot        <>
-      ", issuer: "       <> condenseKey issuer   <>
-      ", delegate: "     <> condenseKey delegate <>
-      ")"
-    where
-      psigCert = CC.Block.delegationCertificate
-               . CC.Block.headerSignature
-               . byronHeader
-               $ hdr
-      issuer   = CC.Delegation.issuerVK   psigCert
-      delegate = CC.Delegation.delegateVK psigCert
-
-      condenseKey :: Crypto.VerificationKey -> String
-      condenseKey = T.unpack . sformat build
-
-      condensedHash
-        = T.unpack
-        . sformat CC.Block.headerHashF
-        . unByronHash
-        . byronHeaderHash
-        $ hdr
-
-      condensedPrevHash
-        = T.unpack
-        . sformat CC.Block.headerHashF
-        . CC.Block.headerPrevHash
-        . byronHeader
-        $ hdr
-
-      condensedSlot
-        = T.unpack
-        . sformat build
-        . unAnnotated
-        . CC.Block.aHeaderSlot
-        . byronHeader
-        $ hdr
-
 {-------------------------------------------------------------------------------
   Epoch Boundary Blocks
 -------------------------------------------------------------------------------}
 
 data ByronBlockOrEBB = ByronBlockOrEBB
   { bbRaw    :: !(CC.Block.ABlockOrBoundary ByteString)
   , bbSlotNo :: !SlotNo
+  , bbHash   :: ByronHash
   } deriving (Eq, Show)
 
 mkByronHeaderOrEBB' :: SlotNo
+                    -> ByronHash
                     -> Either (CC.Block.ABoundaryHeader ByteString)
                               (CC.Block.AHeader         ByteString)
                     -> Header ByronBlockOrEBB
-mkByronHeaderOrEBB' slotNo header = case header of
-    -- By pattern-matching on the two cases, we satisfy the format
-    -- 'ByronHeaderOrEBB' desires.
-    Left ebb -> ByronHeaderBoundary ebb slotNo h
-      where
-        h = ByronHash $ CC.Block.boundaryHeaderHashAnnotated ebb
-    Right mb -> ByronHeaderRegular mb slotNo h
-      where
-        h = ByronHash $ CC.Block.headerHashAnnotated mb
+mkByronHeaderOrEBB' slotNo hdrHash header = case header of
```

Comment:
Assert that slot number explicitly passed matches the one in the (regular) header.

Reply:
Can't. Instead added comment:

```
-- NOTE: The @slotNo@ should correspond to the one that we can compute from the
-- header (using 'computeHeaderSlot') -- except we can't actually /do/ that
-- conversion here since we'd need the know @epochSlots@ for that.
```